pkg/segment/writer/agiletree.go

```diff
@@ -216,6 +222,34 @@ func (stb *StarTreeBuilder) DropSegTree(stbDictEncWorkBuf [][]string) {
 	stb.ResetSegTree(stb.groupByKeys, stb.mColNames, stbDictEncWorkBuf)
 }
 
+func (stb *StarTreeBuilder) DropColumn(colToDrop string) error {
+	dropLevel := -1
+	for idx, col := range stb.groupByKeys {
+		if col == colToDrop {
+			dropLevel = idx
+			continue
+		}
+	}
+	if dropLevel == -1 {
+		return fmt.Errorf("DropColumn: column to drop %v not found", colToDrop)
+	}
+	if stb.droppedCols[dropLevel] {
+		return fmt.Errorf("DropColumn: column %v already dropped", colToDrop)
+	}
+	nextLevel := stb.getNextLevel(-1)
+	if nextLevel == -1 {
+		return fmt.Errorf("DropColumn: cannot drop last column")
+	}
```

Comment:
Why are there restrictions on which columns we can drop?

Reply:
Sorry the comment is incorrect. It means that we don't have any columns to drop.